compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -12357,8 +12361,7 @@ private STNode parseSignedIntOrFloat() {
         return STNodeFactory.createUnaryExpressionNode(operator, literal);
     }
 
-    private boolean isSingletonTypeDescStart(SyntaxKind tokenKind) {
-        STToken nextNextToken = getNextNextToken();
+    static boolean isSingletonTypeDescStart(SyntaxKind tokenKind, STToken nextNextToken) {
```

Comment:
```suggestion
    private static boolean isSingletonTypeDescStart(SyntaxKind tokenKind, STToken nextNextToken) {
```

Reply:
This is called by `isTypeStartingToken` which is a static method. Hence `isSingletonTypeDescStart` has to be static